tests/utils.go

```diff
@@ -100,3 +110,38 @@ func GetKubeClientFromRESTConfig(config *rest.Config) (*kubernetes.Clientset, er
 
 	return coreClient, nil
 }
+
+// Creates a new namespace with a randomly generated name that starts with cdi-test-
+func GenerateNamespace(client *kubernetes.Clientset, baseName string) *k8sv1.Namespace {
+	var namespace *k8sv1.Namespace
+	var err error
+	namespaceName := fmt.Sprintf("cdi-test-%s-%s", baseName, strings.ToLower(util.RandAlphaNum(10)))
+	nsDef := &k8sv1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if wait.PollImmediate(2*time.Second, defaultTimeout, func() (bool, error) {
+		namespace, err = client.CoreV1().Namespaces().Create(nsDef)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	}) != nil {
+		Fail("Unable to create namespace: " + namespaceName)
+	}
+	return namespace
+}
+
+// Destroys the passed in name space, be sure to clean any resources used before destroying the namespace.
+func DestroyNamespace(client *kubernetes.Clientset, namespace *k8sv1.Namespace) {
```

Comment:
we might want a DestroyNamespaceAll() function as well that runs before the testsuite starts and after the testsuite ends.  This would ensure that we don't have namespaces left over in the event that we "Ctrl-C' a functional test to stop it early.

Reply:
True. k8s e2e has similar but with predefined namespaces that they don't want left over.

Reply:
Would it make sense to pop the namespace in a global list when we create it, and remove it when we delete the namespace, then in the after suite loop over the list and if anything is there we can delete at that point.

Reply:
I think this "global list" can be the _framework_ struct/object. So the `AfterSuite` can clean up the resources in the framework as a final check to make sure the test has fully cleaned up after it's done. This is important once you start running these tests in parallel.